src/components/TableHeadCell.js

```diff
@@ -116,11 +116,14 @@ const TableHeadCell = ({
     item: {
       type: 'HEADER',
       colIndex: index,
+      headCellRefs: draggableHeadCellRefs,
     },
     begin: monitor => {
-      setHintTooltipOpen(false);
-      setSortTooltipOpen(false);
-      setDragging(true);
+      setTimeout(() => {
+        setHintTooltipOpen(false);
+        setSortTooltipOpen(false);
+        setDragging(true);
+      }, 0);
```

Comment:
Can you tell what this is❓ 

why use setTimeout..

ref : https://react-dnd.github.io/react-dnd/docs/api/use-drag

begin(monitor): Optional. Fired when a drag operation begins. Nothing needs to be returned, but if an object is returned it will override the default item property of the spec.

Reply:
An attempt to solve https://github.com/gregnb/mui-datatables/issues/1359 - I cannot reproduce that issue. I found this thread: https://github.com/react-dnd/react-dnd/issues/1085 - it had a workaround that seemed related to what the guy was describing. Wrapping in setTimeout doesn't really have ill effects, it's just odd. I also can't reproduce https://github.com/gregnb/mui-datatables/issues/1363. I tried creating an example with other react-dnd componets but everything seems to work fine. I found this issue: https://github.com/react-dnd/react-dnd/issues/1571 and it had a harmless solution so I added that in. Hopefully that will resolve the problem that guy is seeing.

Reply:
@patorjk 

Thanks for the kind explanation.

Reply:
It would be okay to change to `Immediately-invoked function expression`.

I'll test it and tell you again